test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Reader/JsonLight/PropertyAndValueJsonLightReaderIntegrationTests.cs

```diff
@@ -353,6 +367,64 @@ public void ReadingTypeDefinitionPayloadJsonLight()
             address.Properties.FirstOrDefault(s => string.Equals(s.Name, "CountryRegion", StringComparison.OrdinalIgnoreCase)).Value.Should().Be("China");
         }
 
+        [Fact]
+        public void ReadingTypeDefinitionPayloadJsonLightWithOmittedNullValues()
+        {
+            EdmEntityType entityType;
+            EdmEntitySet entitySet;
+            EdmModel model = BuildEdmModelForOmittedNullValuesTestCases(out entityType, out entitySet);
+
+            const string payload = @"
+{
+                ""@odata.context"":""http://www.example.com/$metadata#EntityNs.MyContainer.People/$entity"",
+                ""@odata.id"":""http://mytest"",
+                ""Id"":0,
+                ""Weight"":60.5,
+                ""Address"":{""CountryRegion"":""US"", ""City"":""Redmond""}
+}";
+
+            List<ODataResource> entries = new List<ODataResource>();
+            ODataNestedResourceInfo navigationLink = null;
+            this.ReadEntryPayload(model, payload, entitySet, entityType,
+                reader =>
+                {
+                    switch (reader.State)
+                    {
+                        case ODataReaderState.ResourceStart:
+                            entries.Add(reader.Item as ODataResource);
+                            break;
+                        case ODataReaderState.NestedResourceInfoStart:
+                            navigationLink = (ODataNestedResourceInfo)reader.Item;

```

Comment:
rename navigationLink to "nestedResourceInfo".

Reply:
Note that "navigationLink" is still used for nestedResourceInfo in a number of places

---
In reply to: [173003599](https://github.com/OData/odata.net/pull/1106#discussion_r173003599) [](ancestors = 173003599)

Reply:
I think you are looking at some original test cases that are not changed by this PR. Since you have spotted them, and it makes sense to renaming it, I have updated the names to 'nestedResourceInfo'.

---
In reply to: [233550345](https://github.com/OData/odata.net/pull/1106#discussion_r233550345) [](ancestors = 233550345,173003599)